### PR TITLE
Improve migration code to support old and new visibility logic

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/resources/liquibase/db-changelog-2.0.xml [hunk 1]
```diff
@@ -446,4 +446,18 @@
...
     </changeSet>
 
+    <changeSet id="set-visibility-migration-true-on-repositories" author="anandJ">
+        <update tableName="repository">
+            <column name="visibility_migration" valueBoolean="false"/>
+            <where>visibility_migration=true</where>
+        </update>
+    </changeSet>
+
+    <changeSet id="set-visibility-migration-true-on-projects" author="anandJ">
```
Comment (conradoverta) [inline backend/src/main/resources/liquibase/db-changelog-2.0.xml]:
```suggestion
    <changeSet id="set-visibility-migration-false-on-projects" author="anandJ">
```

File: backend/src/main/resources/liquibase/db-changelog-2.0.xml [hunk 2]
```diff
@@ -446,4 +446,18 @@
         <tagDatabase tag="db_version_2.20"/>
     </changeSet>
 
+    <changeSet id="set-visibility-migration-true-on-repositories" author="anandJ">
```
Comment (conradoverta) [inline backend/src/main/resources/liquibase/db-changelog-2.0.xml]:
```suggestion
    <changeSet id="set-visibility-migration-false-on-repositories" author="anandJ">
```

File: backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java [hunk 3]
```diff
@@ -97,35 +100,63 @@ private static void migrateProjects() {
...
+                      .build(),
+                  getResourceVisibility(
+                      Optional.ofNullable(
+                          WorkspaceTypeEnum.WorkspaceType.forNumber(project.getWorkspace_type())),
+                      VisibilityEnum.Visibility.forNumber(project.getProject_visibility())));
+            } else {
+              GetResourcesResponseItem resourceDetails = responseItemMap.get(project.getId());
+              roleService.createWorkspacePermissions(
+                  resourceDetails.getWorkspaceId(),
+                  Optional.empty(),
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java:152]: This argument is new compared to the baseline above. What is it? I tend to get confused with overloaded methods =)

**Review by conradoverta — APPROVED** [review]
One question, but LGTM

File: backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java (hunk 3 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java:152]: it is unused argument of `createWorkspacePermissions` which i will remove in follow up PR. baseline above method refactor already so it is not there. Sorry for confusion.

File: backend/src/main/resources/liquibase/db-changelog-2.0.xml (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/resources/liquibase/db-changelog-2.0.xml]: Fixed

**Pull request merged**